test/ctest_helpers.cmake

```diff
@@ -133,3 +142,125 @@ function(create_test_executable target)
         add_dependencies(${target} copy_dir_${target}_${dir_basename})
     endforeach()
 endfunction()
+
+function(define_test_variants target)
+    cmake_parse_arguments(
+        PARSE_ARGV
+        1
+        "DEFINE_TEST"
+        ""
+        ""
+        "VARIANTS;LABELS"
+    )
+    if (NOT "${DEFINE_TEST_UNPARSED_ARGUMENTS}" STREQUAL "")
+        message(FATAL_ERROR "Unknown arguments to define_test_variants: ${DEFINE_TEST_VARIANTS_UNPARSED_ARGUMENTS}")
+    endif()
+    if ("${DEFINE_TEST_VARIANTS}" STREQUAL "")
+        message(FATAL_ERROR "Need at least one variant for define_test_variants")
+    endif()
+
+    set(defined_tests)
+    foreach(variant ${DEFINE_TEST_VARIANTS})
+        list(LENGTH variant variant_length)
+        if (NOT variant_length EQUAL 2)
+            message(
+                FATAL_ERROR
+                "Invalid variant format: ${variant} - Expected format 'variant_name;variant args'"
+            )
+        endif()
+        list(GET variant 0 curr_variant_name)
+        list(GET variant 1 curr_variant_args)
+        set(variant_args)
+        if(WT_WIN)
```

Comment:
Where is WT_WIN being set? I'm not that familiar with cmake

Reply:
Yep, good question, its no immediately obvious! 
It gets set when the Windows CMake config file gets included here: 
https://github.com/wiredtiger/wiredtiger/blob/6a8fcdfcfe7815cfcd5350ad86f3dd68c7e99cee/build_cmake/configs/x86/windows/config.cmake#L10
`WT_OS` is a `config_choice` option (a generic helper we defined), that bounds the values of `WT_OS` to `darwin`, `windows` or `linux`, defined here:
https://github.com/wiredtiger/wiredtiger/blob/6a8fcdfcfe7815cfcd5350ad86f3dd68c7e99cee/build_cmake/configs/base.cmake#L23-L30
Depending on the value of `WT_OS` an additional variable will be defined, being either `WT_DARWIN`, `WT_WIN` or `WT_LINUX`. Its mostly there for convenience so as to avoid defining `"${WT_OS}" STREQUAL "windows"` all over the place.

Reply:
Ah okay that explains why I was confused. 👍